### PR TITLE
add ExpandableGeneric filter type

### PR DESCRIPTION
**Pull request by dshafer**
- [x] Update CHANGELOG.md
- [x] Update README.md (as needed)
- [x] New dependencies were added to `pyproject.toml`
- [x] `pip install` succeeds with a clean virtualenv
- [x] There are new or modified tests that cover changes
- [x] Test coverage is maintained or expanded

### CODE REVIEW

**Comment by bgrant** [comment]
Hey @dshafer - can you tell me the context around this?  What's an ExpandableGeneric, and why does it need different handling than other expandables?

**Comment by dshafer** [comment]
The original `ExpandableForeignKey` is (as the name suggests) very much tied to the expandable field being a Django ORM foreign key object - that is, if the field is named `foo`, there's another hidden field `foo_id` that holds the primary key of the expandable object, and this `foo_id` is used to build the non-expanded representation of the object.

`ExpandableGeneric` allows expandables to be any type, and for filters to explicitly define the non-expanded view of the object.  This is needed as we transition away from the Django ORM, but still want to offer the ability to return compact representations of nested objects in responses.

The particular need for this change to happen now is that the upcoming change to FakeForeignKey in our service no longer makes it behave exactly like a Django foreign key when it's linking to something in a non-ORM datastore, so we need more control to be able to specify the collapsed format

File: tests/test_filters.py [hunk 1]
```diff
@@ -31,6 +31,23 @@ def setUp(self):
...
+    def test_expandable_generic_field(self):
+        # test unexpanded
+        filtered = filtering.apply_filters_to_object(
+            self.test_model, filters.DEFAULT_FILTERS
+        )
+        self.assertIn("expandable_generic", filtered)
+        self.assertEqual("1234", filtered["expandable_generic"]["id"])
+        self.assertNotIn("expanded", filtered["expandable_generic"])
+
+        # test exexpanded
```
Comment (bgrant) [inline tests/test_filters.py:43]: typo: `exexpanded` -> `expanded`

**Review by bgrant — APPROVED** [review]
One typo fix requested.

**Review by bgrant — APPROVED** [review]

**Pull request merged**